realm/realm-library/src/objectServer/java/io/realm/internal/mongodb/Request.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.mongodb;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import javax.annotation.Nullable;
+
+import io.realm.RealmAsyncTask;
+import io.realm.internal.RealmNotifier;
+import io.realm.internal.android.AndroidCapabilities;
+import io.realm.internal.android.AndroidRealmNotifier;
+import io.realm.internal.async.RealmAsyncTaskImpl;
+import io.realm.log.RealmLog;
+import io.realm.mongodb.App;
+import io.realm.mongodb.ErrorCode;
+import io.realm.mongodb.ObjectServerError;
+
+// Class wrapping requests made against MongoDB Realm. Is also responsible for calling with success/error on the
+// correct thread.
+public abstract class Request<T> {
+    @Nullable
+    private final App.Callback<T> callback;
+    private final RealmNotifier handler;
+    private final ThreadPoolExecutor networkPoolExecutor;
+
+    // FIXME Made public to use in Functions. Consider reworking when App, User is moved
```

Comment:
Since this now lives in internal, it is perfectly fine to have public constructors. So this FIXME can probably just be removed